build.gradle.kts

```
@@ -16,7 +16,10 @@ apply(plugin = "com.github.sherter.google-java-format")
 
 allprojects {
     group = "com.newrelic.telemetry"
-    version = project.findProperty("releaseVersion") as String
+    // -Prelease=true will render a non-snapshot version
+// All other values (including unset) will render a snapshot version.
+    val release: String? by project
+    version = "0.6.1" + if("true" == release) "" else "-SNAPSHOT"
```

this version feels very hidden down in the guts of the gradle script. Can we make it more prominent and obvious so it's very clear where to find it when we want to change it?

---

Thanks for calling that out. Would this larger comment suffice or would you prefer a method call here?

```suggestion
/* Setting version for release or snapshot
     *
     * Passing in -Prelease=true will render a non-snapshot version
     *
     * All other values (including unset) will render a snapshot version.
     */
     
    version = "0.6.1" + if("true" == release) "" else "-SNAPSHOT"
```

---

Can we make the version a property at the top (or near the top) of the build script, or some such? I don't like having the only reference embedded in a line of kotlin that makes it harder to see.

---

I'll take a look

I tried moving it to the top, out of the `allprojects {}` closure, but when the script is resolved, each individual project received `unspecified` as their version.

---

can you at least move it to the top of the `allProjects` with a name like `baseVersion`?

---

Ok, it was more obvious than I thought.   It's basically how it was before.  Just change the version in the gradle.properties